### PR TITLE
perf: serialization optimization

### PR DESCRIPTION
**Pull request by larme**
## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/_internal/runner/container.py [hunk 1]
```diff
@@ -290,13 +291,36 @@ def to_payload(
...
 
+        import numpy as np
+        # skip 0-dimensional array
+        if batch.shape:
+
+            buffers: list[pickle.PickleBuffer] = []
+            if not (batch.flags["C_CONTIGUOUS"] or batch.flags["F_CONTIGUOUS"]):
+                # TODO: use fortan contiguous if it's faster
+                batch = np.ascontiguousarray(batch)
+            bs = pickle.dumps(batch, protocol=5, buffer_callback=buffers.append)
```
Comment (aarnphm) [inline src/bentoml/_internal/runner/container.py:305]: we probably need backport for pickle5

**Comment by larme** [comment]
serve as a base of discussion now, no need to review

File: src/bentoml/_internal/runner/container.py [hunk 2]
```diff
@@ -290,13 +291,36 @@ def to_payload(
             return cls.create_payload(
                 plasma_db.put(batch).binary(),
                 batch.shape[batch_dim],
-                {"plasma": True},
+                {"format": "plasma"},
```
Comment (aarnphm) [inline src/bentoml/_internal/runner/container.py:294]: Let's also remove all of the plasmadb here as well, since the API is going to be removed soon.

**Review by aarnphm — COMMENTED** [review]
quick initial read.

File: src/bentoml/_internal/runner/container.py [hunk 3]
```diff
@@ -287,13 +288,36 @@ def to_payload(
...
+            )
+
+        import numpy as np
```
Comment (aarnphm) [inline src/bentoml/_internal/runner/container.py:294]: can we do `LazyLoader` here to L33?

File: src/bentoml/_internal/runner/container.py (hunk 2 above)
Comment (larme) [reply src/bentoml/_internal/runner/container.py]: done

File: src/bentoml/_internal/runner/container.py [hunk 4]
```diff
@@ -23,6 +30,7 @@
 
     from .. import external_typing as ext
```
Comment (aarnphm) [inline src/bentoml/_internal/runner/container.py:31]:
```suggestion
    import numpy as np
    
    from .. import external_typing as ext
```

File: tests/integration/frameworks/models/tensorflow.py [hunk 5]
```diff
@@ -286,6 +286,11 @@ def make_keras_functional_model() -> tf.keras.Model:
         make_keras_functional_model(),
         make_keras_sequential_model(),
     ]
+    for inp in [
+        input_tensor,
+        input_array,
+        input_data,
+    ]
```
Comment (aarnphm) [inline tests/integration/frameworks/models/tensorflow.py:293]: actually this fixes the CI right? should we split this into a separate PR?

**Review by bojiang — APPROVED** [review]

**Pull request merged**